### PR TITLE
net.replicaset: fix a flaky test

### PR DESCRIPTION
**Pull request by alyapunov**
Fix several problems that led to unexpected results:
* Handle error thrown by box.clt.promote. Not very often but it could fail ruining the test. Fixed by retry.
* It's not so obvious but RO server after successful call of box.ctl.promote for some time remains RO. On the other hand both net.replicaset and the test rely box.info.ro to define the leader, for in their view the actual switch to the new leader happens some (perhaps different) time after call to box.ctl.promote. That asynchrony sometimes led to unexpected results. Fixed by waiting of RW status.
* The fix above (wait for RW of the new leader) introduced new problem: in case with sequential leader change and broadcast of new values, the new leader has good chances (but not 100%) to deliver old broadcasted value to the watcher. Fix it by splitting the sequence to the leader change with verification and broadcast with verification.

Follow-up #9823

NO_DOC=fix flaky test
NO_CHANGELOG=fix flaky test

### CODE REVIEW

File: test/box-luatest/gh_9823_net_replicaset_test.lua [hunk 1]
```diff
@@ -322,7 +325,10 @@ g.test_watch_leader = function(cg)
     -- Change the leader.
```
Comment (sergepetrenko) [inline test/box-luatest/gh_9823_net_replicaset_test.lua:325]: Found another flaky fail when testing locally:
```
[002] not ok 4T>box-luatest.gh_9823_net_replicaset.test_watch_leader                                                           
[002] #   ...e/sptnt/test/box-luatest/gh_9823_net_replicaset_test.lua:295: expected: 1, actual: 2
[002] #   stack traceback: 
```

**Review by sergepetrenko — COMMENTED** [review]
Thanks for the patch!

File: test/box-luatest/gh_9823_net_replicaset_test.lua (hunk 1 above)
Comment (alyapunov) [reply test/box-luatest/gh_9823_net_replicaset_test.lua:336]: Added another fix.

Comment (sergepetrenko) [reply test/box-luatest/gh_9823_net_replicaset_test.lua:336]: Thanks! I can't reproduce the failure anymore.

**Review by sergepetrenko — APPROVED** [review]
Thanks for the fix!

**Pull request merged**